src/main.js

```diff
@@ -167,6 +178,95 @@ function onPageChange() {
       subtree: true,
     });
     console.info("Monitoring", [feed]);
+    return;
+  }
+
+  // FB5 design
+  // there's a feed div that we don't monitor yet
+  feed = document.querySelector("div[role=feed]:not([data-adblock-monitored])");
+  if (feed !== null) {
+    setFeedObserver();
+    return;
+  }
+  // there's a feed loading placeholder
+  feed = document.querySelector("div#suspended-feed");
+  if (feed !== null) {
+    setFeedObserver();
+    return;
+  }
+  // No new feed was detected
+  // Cleanup observer when there's no feed monitored left in DOM
+  if (feedObserver !== null && document.querySelector("div[role=feed][data-adblock-monitored]") === null) {
+    feedObserver.disconnect();
+  }
+}
+
+// wait for and observe FB5 feed element
+function setFeedObserver() {
+  // We are expecting to find a new feed div
+  let feed = document.querySelector("div[role=feed]:not([data-adblock-monitored])");
+  if (feed !== null) {
+    // check existing posts
+    feed
+      .querySelectorAll('div[data-pagelet^="FeedUnit_"]')
+      .forEach(hideIfSponsored);
+
+    let feedContainer = feed.parentNode;
+    // flag this feed as monitored
+    feed.dataset.adblockMonitored = true;
+    feedObserver = new MutationObserver((mutations) => {
+      mutations.forEach((mutation) => {
+        // check if feed was reloaded without changing page
+        if (mutation.target === feedContainer && mutation.addedNodes.length > 0) {
+          feedObserver.disconnect();
+          // check again for the new feed, after the current queue and DOM is done 
+          setTimeout(setFeedObserver, 0);
```

Comment:
Not sure if I understand this?

Reply:
when the user is at the new home feed page and clicks to go to the home page (either from the FB logo or the home icon) the div[role=feed] is being destroyed and created again. Here we get the notification for such event and instead of directly start looking for the element to hook the observer again, I use this trick to put our method at the end of the current queue stack allowing the browser to handle and render the new div before we start looking for elements. Basically i'm buying some time... you can find more about this searching for "event loop" or [this video](https://www.youtube.com/watch?v=cCOL7MC4Pl0)

Reply:
fixed